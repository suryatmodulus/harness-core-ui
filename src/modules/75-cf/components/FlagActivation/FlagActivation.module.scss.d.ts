/* eslint-disable */
/**
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 **/
// this is an auto-generated file, do not update this manually
declare const styles: {
  readonly actionButtons: string
  readonly btnCode: string
  readonly contentHeading: string
  readonly editBtnsGroup: string
  readonly editContainer: string
  readonly envSelect: string
  readonly formContainer: string
  readonly noEdit: string
  readonly switch: string
  readonly tabContainer: string
  readonly tabTitle: string
  readonly testFlagDialog: string
  readonly testFlagDialogContainer: string
}
export default styles
