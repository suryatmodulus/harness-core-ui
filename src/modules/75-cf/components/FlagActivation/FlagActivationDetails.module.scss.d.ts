/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

/* eslint-disable */
// this is an auto-generated file
declare const styles: {
  readonly editFlagModal: string
  readonly variationsContainer: string
  readonly tagsFlagActivationDetails: string
  readonly newEnvRulesContainer: string
  readonly selectEnv: string
  readonly editDetailsModalContainer: string
  readonly checkboxEditDetails: string
  readonly collapseFeatures: string
  readonly collapseFeaturesPrerequisites: string
  readonly variationsList: string
  readonly variationItem: string
  readonly module: string
  readonly breadcrumb: string
}
export default styles
