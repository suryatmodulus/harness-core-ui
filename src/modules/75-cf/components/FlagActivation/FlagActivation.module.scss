/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

.env-select {
  width: 200px !important;
}

.test-flag-dialog {
  width: 1000px;
  height: 600px;
  position: relative;
  padding: 0;
  .test-flag-dialog-container {
    height: 100%;
    padding: var(--spacing-medium) var(--spacing-xlarge) !important;
    [class*='OverlaySpinner'] {
      height: inherit;
      overflow: auto;
      & > form {
        height: inherit;
        & > div {
          height: inherit;
        }
      }
    }
  }
}

.tabContainer {
  position: relative;
  height: calc(100% - 74px - 82px);
  overflow: auto;

  &.noEdit {
    height: calc(100% - 74px);
  }

  :global(.bp3-tab-list) {
    border-bottom: 1px solid #e4e4eb !important;
    padding-left: var(--spacing-huge);

    [class='bp3-tab'] {
      margin-top: var(--spacing-medium);
      margin-right: var(--spacing-medium) !important;
    }
  }

  .btnCode {
    position: absolute;
    right: 15px;
    top: 13px;
  }
}

.edit-container {
  overflow: auto;
  .edit-btns-group {
    position: fixed;
    bottom: 0;
    width: 100% !important;
    border: 1px solid var(--grey-300);
    padding: var(--spacing-large) 0 !important;
    background-color: var(--white);
  }
}

.contentHeading {
  .switch {
    margin-bottom: 0 !important;
    margin-left: -10px;
  }

  :global {
    .bp3-control.bp3-switch input:checked ~ .bp3-control-indicator {
      background-color: #25a6f7;
    }
  }
}

.formContainer {
  position: relative;
  height: 100vh;

  .actionButtons {
    position: absolute;
    bottom: 0;
    right: 0;
    width: 100% !important;
    border-top: 1px solid #e4e4eb;
    padding-left: var(--spacing-huge) !important;
    padding-top: 22px !important;
    height: 80px;
    background-color: var(--white) !important;
  }
}

.tabTitle {
  font-size: 14px !important;
  font-weight: 400 !important;
  line-height: 19px !important;
  color: #333333 !important;
  padding: var(--spacing-small) var(--spacing-xsmall) var(--spacing-xsmall) !important;
}

[aria-selected='true'] > .tabTitle {
  font-weight: 600 !important;
  color: #22272d !important;
}
