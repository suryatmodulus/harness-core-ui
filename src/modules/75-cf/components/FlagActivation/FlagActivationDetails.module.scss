/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

.edit-flag-modal {
  width: fit-content !important;
  max-width: 90vw;
}

.variations-container {
  & > div {
    margin-right: var(--spacing-medium);
  }
}

.tags-flag-activation-details {
  margin-top: var(--spacing-medium) !important;

  & > p {
    display: inline-block;
    border-radius: var(--spacing-small);
  }
}

.new-env-rules-container {
  align-items: baseline;
  .select-env {
    margin-right: var(--spacing-medium) !important;
  }
}

.edit-details-modal-container {
  padding: 0 var(--spacing-xxlarge) var(--spacing-xlarge) !important;
  & > p {
    margin-bottom: var(--spacing-large) !important;
    font-size: var(--font-size-medium) !important;
    color: var(--black) !important;
    font-weight: bold !important;
  }
  .checkbox-edit-details {
    margin-right: var(--spacing-xsmall) !important;
    :global(.bp3-control-indicator) {
      margin-left: 0 !important;
    }
  }
}

.collapse-features {
  margin: var(--spacing-medium) 0 !important;
  padding: var(--spacing-xsmall) var(--spacing-small) !important;

  & > div {
    border: 0 !important;
    padding: var(--spacing-small) !important;
  }

  [class*='CollapseHeader---leftSection'] {
    span {
      color: var(--primary-7) !important;
    }
  }

  [class*='Collapse--main'] [class*='Collapse--header'] {
    padding-top: var(--spacing-xsmall) !important;
  }

  .collapse-features-prerequisites {
    padding: var(--spacing-xsmall) !important;
    border: 1px solid var(--grey-200);
    & > button {
      padding-left: 0;
      align-self: start;
    }
  }
}

.variationsList {
  .variationItem {
    padding-top: var(--spacing-medium) !important;
  }
}

.module {
  background: var(--white) !important;
  box-shadow: 0px 0px 1px rgba(40, 41, 61, 0.04), 0px 2px 4px rgba(96, 97, 112, 0.16);
  border-radius: 4px;
}

.breadcrumb {
  margin-bottom: -12px !important;

  :global {
    .bp3-popover-wrapper {
      transform: translate(12px, -10px);
    }
  }
}
