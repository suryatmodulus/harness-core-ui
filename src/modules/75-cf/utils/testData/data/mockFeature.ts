/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { Feature } from 'services/cf'

const mockFeature: Feature = {
  archived: false,
  createdAt: 1635243729074,
  defaultOffVariation: 'false',
  defaultOnVariation: 'true',
  description: '',
  envProperties: {
    defaultServe: { variation: 'false' },
    environment: 'testnonprod',
    modifiedAt: 1635333973373,
    offVariation: 'false',
    rules: [],
    state: 'on',
    variationMap: [
      {
        targetSegments: ['test_target_group'],
        targets: [{ identifier: 'another_target', name: 'another target' }],
        variation: 'true'
      }
    ],
    version: 56
  },
  evaluation: 'false',
  evaluationIdentifier: 'false',
  identifier: 'new_flag',
  kind: 'boolean',
  modifiedAt: 1635333973371,
  name: 'new flag ',
  owner: ['chris.blakely@harness.io'],
  permanent: false,
  prerequisites: [],
  project: 'chrisgit2',
  results: undefined,
  status: { lastAccess: -6795364578871, status: 'never-requested' },
  tags: [],
  variations: [
    { identifier: 'true', name: 'True', value: 'true' },
    { identifier: 'false', name: 'False', value: 'false' }
  ]
}

export default mockFeature
