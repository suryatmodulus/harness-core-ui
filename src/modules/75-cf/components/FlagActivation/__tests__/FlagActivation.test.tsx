/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render, RenderResult, screen, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { TestWrapper } from '@common/utils/testUtils'
import * as cfServiceMock from 'services/cf'

import mockFeature from '@cf/utils/testData/data/mockFeature'
import mockGitSync from '@cf/utils/testData/data/mockGitSync'
import FlagActivation from '../FlagActivation'

jest.mock('services/cf')

jest.mock('@cf/hooks/useEnvironmentSelectV2', () => ({
  useEnvironmentSelectV2: jest.fn().mockReturnValue({
    data: [],
    loading: false,
    error: undefined,
    refetch: jest.fn(),
    EnvironmentSelect: function EnvironmentSelect() {
      return <div />
    }
  })
}))

const renderComponent = (): RenderResult => {
  return render(
    <TestWrapper
      path="/account/:accountId/cf/orgs/:orgIdentifier/projects/:projectIdentifier/feature-flags"
      pathParams={{ accountId: 'dummy', orgIdentifier: 'dummy', projectIdentifier: 'dummy' }}
    >
      <FlagActivation gitSync={mockGitSync} flagData={mockFeature} project="chris_test" refetchFlag={jest.fn()} />
    </TestWrapper>
  )
}

describe('FlagActivation', () => {
  beforeEach(() => {
    jest.clearAllMocks()

    jest
      .spyOn(cfServiceMock, 'useGetAllFeatures')
      .mockReturnValue({ data: [], loading: false, mutate: jest.fn() } as any)
    jest
      .spyOn(cfServiceMock, 'useGetTargetsAndSegmentsInfo')
      .mockReturnValue({ data: [], loading: false, mutate: jest.fn() } as any)
    jest
      .spyOn(cfServiceMock, 'useGetAllSegments')
      .mockReturnValue({ data: [], loading: false, mutate: jest.fn() } as any)
    jest
      .spyOn(cfServiceMock, 'useGetAllTargetAttributes')
      .mockReturnValue({ data: [], loading: false, mutate: jest.fn() } as any)
    jest
      .spyOn(cfServiceMock, 'useGetFeatureEvaluations')
      .mockReturnValue({ data: [], loading: false, mutate: jest.fn() } as any)
    jest.spyOn(cfServiceMock, 'usePatchFeature').mockReturnValue({ data: [], loading: false, mutate: jest.fn() } as any)
    jest
      .spyOn(cfServiceMock, 'useGetAllTargets')
      .mockReturnValue({ data: [], loading: false, mutate: jest.fn() } as any)
  })

  test('it should render correctly', async () => {
    const { container } = renderComponent()

    expect(container).toMatchSnapshot()
  })

  test('it should render form correctly', async () => {
    const patchMock = jest.fn()
    jest.spyOn(cfServiceMock, 'usePatchFeature').mockReturnValue({ data: [], loading: false, mutate: patchMock } as any)

    renderComponent()

    // Edit flag variation
    await waitFor(() => expect(screen.getByText('cf.featureFlags.rules.editRules')).toBeInTheDocument())

    userEvent.click(screen.getByText('cf.featureFlags.rules.editRules'))

    expect(document.getElementsByName('onVariation')[0]).toBeInTheDocument()

    userEvent.click(document.getElementsByName('onVariation')[0])
    userEvent.click(screen.getByText('False'))

    expect(document.getElementsByName('onVariation')[0]).toHaveValue('False')
    expect(screen.getByText('save')).toBeInTheDocument()

    // Assert modal appears
    userEvent.click(screen.getByText('save'))

    const saveToGitModal = document.getElementById('save-flag-to-git-modal-body')
    await waitFor(() => expect(saveToGitModal).toBeInTheDocument())

    expect(saveToGitModal).toMatchSnapshot()
  })
})
