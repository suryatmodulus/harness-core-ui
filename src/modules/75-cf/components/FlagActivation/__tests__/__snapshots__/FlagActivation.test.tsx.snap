// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlagActivation it should render correctly 1`] = `
<div>
  <div
    class="OverlaySpinner--overlaySpinner"
  >
    <form
      class="FormikForm--main"
      errors="[object Object]"
      initialvalues="[object Object]"
      submitcount="0"
      touched="[object Object]"
      values="[object Object]"
    >
      <div>
        <div
          class="StyledProps--font StyledProps--main formContainer"
        >
          <div
            class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--padding-large"
            style="background-color: rgb(244, 246, 255); mix-blend-mode: normal; box-shadow: 0px 0px 1px rgba(40, 41, 61, 0.04), 0px 2px 4px rgba(96, 97, 112, 0.16);"
          >
            <span
              style="flex-grow: 1;"
            />
            <div
              class="Layout--horizontal Layout--layout-spacing-small StyledProps--main container"
            >
              <span
                class="label"
              >
                environment
              </span>
              <div />
            </div>
          </div>
          <div
            class="StyledProps--font StyledProps--main tabContainer noEdit"
          >
            <div
              class="bp3-tabs Tabs--main Tabs--horizontal"
            >
              <div
                class="bp3-tab-list"
                role="tablist"
              >
                <div
                  class="bp3-tab-indicator-wrapper"
                  style="height: 0px; transform: translateX(0px) translateY(0px); width: 0px;"
                >
                  <div
                    class="bp3-tab-indicator"
                  />
                </div>
                <div
                  aria-controls="bp3-tab-panel_editFlag_targeting"
                  aria-disabled="false"
                  aria-expanded="true"
                  aria-selected="true"
                  class="bp3-tab"
                  data-tab-id="targeting"
                  id="bp3-tab-title_editFlag_targeting"
                  role="tab"
                  tabindex="0"
                >
                  <p
                    class="StyledProps--font StyledProps--main tabTitle"
                  >
                    cf.featureFlags.targeting
                  </p>
                </div>
                <div
                  aria-controls="bp3-tab-panel_editFlag_metrics"
                  aria-disabled="false"
                  aria-expanded="false"
                  aria-selected="false"
                  class="bp3-tab"
                  data-tab-id="metrics"
                  id="bp3-tab-title_editFlag_metrics"
                  role="tab"
                  tabindex="0"
                >
                  <p
                    class="StyledProps--font StyledProps--main tabTitle"
                  >
                    cf.featureFlags.metrics.title
                  </p>
                </div>
                <div
                  aria-controls="bp3-tab-panel_editFlag_activity"
                  aria-disabled="false"
                  aria-expanded="false"
                  aria-selected="false"
                  class="bp3-tab"
                  data-tab-id="activity"
                  id="bp3-tab-title_editFlag_activity"
                  role="tab"
                  tabindex="0"
                >
                  <p
                    class="StyledProps--font StyledProps--main tabTitle"
                  >
                    cf.featureFlags.activity
                  </p>
                </div>
              </div>
              <div
                aria-hidden="false"
                aria-labelledby="bp3-tab-title_editFlag_targeting"
                class="bp3-tab-panel"
                id="bp3-tab-panel_editFlag_targeting"
                role="tabpanel"
              >
                <div
                  class="Layout--vertical Layout--layout-spacing-medium StyledProps--main StyledProps--padding-left-huge StyledProps--padding-right-large StyledProps--padding-bottom-large"
                >
                  <div
                    class="Layout--horizontal StyledProps--main contentHeading StyledProps--flex StyledProps--flex-alignItems-center"
                  >
                    <p
                      class="StyledProps--font StyledProps--main"
                    >
                      <label
                        class="bp3-control bp3-switch bp3-align-right bp3-large switch"
                      >
                        <input
                          checked=""
                          data-testid="flag-status-switch"
                          type="checkbox"
                        />
                        <span
                          class="bp3-control-indicator"
                        />
                      </label>
                    </p>
                    <p
                      class="StyledProps--font StyledProps--main StyledProps--padding-left-small"
                      style="font-size: 12px; color: rgb(107, 109, 133);"
                    >
                      cf.featureFlags.flagOn
                    </p>
                    <span
                      style="flex-grow: 1;"
                    />
                    <button
                      class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color Button--withLeftIcon"
                      type="button"
                    >
                      <span
                        class="bp3-icon bp3-icon-edit StyledProps--main StyledProps--padding-right-xsmall"
                        icon="edit"
                      >
                        <svg
                          data-icon="edit"
                          height="16"
                          viewBox="0 0 16 16"
                          width="16"
                        >
                          <desc>
                            edit
                          </desc>
                          <path
                            d="M3.25 10.26l2.47 2.47 6.69-6.69-2.46-2.48-6.7 6.7zM.99 14.99l3.86-1.39-2.46-2.44-1.4 3.83zm12.25-14c-.48 0-.92.2-1.24.51l-1.44 1.44 2.47 2.47 1.44-1.44c.32-.32.51-.75.51-1.24.01-.95-.77-1.74-1.74-1.74z"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                      <span
                        class="bp3-button-text"
                      >
                        cf.featureFlags.rules.editRules
                      </span>
                    </button>
                  </div>
                  <div
                    class="Layout--vertical Layout--layout-spacing-medium StyledProps--main"
                  >
                    <h3
                      class="StyledProps--font StyledProps--font-h3 StyledProps--main StyledProps--font-variation-h6"
                      level="3"
                    >
                      <span
                        class="align"
                        data-tooltip-id="ff_ffDefaultRules_heading"
                      >
                        cf.featureFlags.rules.defaultRules
                      </span>
                    </h3>
                    <div
                      class="rules"
                    >
                      <p
                        class="StyledProps--font StyledProps--main"
                      >
                        <span>
                          cf.featureFlags.ifFlagOnServe
                        </span>
                      </p>
                      <div
                        class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main StyledProps--flex StyledProps--flex-alignItems-center StyledProps--flex-justifyContent-flex-start"
                      >
                        <span
                          style="border-radius: 50%; width: 12px; height: 12px; background-color: rgb(101, 222, 242); display: inline-block;"
                        />
                        <span
                          class="StyledProps--font StyledProps--main StyledProps--inline StyledProps--margin-bottom-xsmall"
                          style="margin-bottom: 0px; font-weight: 600;"
                        >
                          False
                        </span>
                      </div>
                      <p
                        class="StyledProps--font StyledProps--main"
                      >
                        <span>
                          cf.featureFlags.ifFlagOffServe
                        </span>
                      </p>
                      <div
                        class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main StyledProps--flex StyledProps--flex-alignItems-center StyledProps--flex-justifyContent-flex-start"
                      >
                        <span
                          style="border-radius: 50%; width: 12px; height: 12px; background-color: rgb(101, 222, 242); display: inline-block;"
                        />
                        <span
                          class="StyledProps--font StyledProps--main StyledProps--inline StyledProps--margin-bottom-xsmall"
                          style="margin-bottom: 0px; font-weight: 600;"
                        >
                          False
                        </span>
                      </div>
                    </div>
                    <p
                      class="StyledProps--font StyledProps--main ruleTitle custom"
                    >
                      <span
                        class="align"
                        data-tooltip-id="ff_ffCustomRules_heading"
                      >
                        cf.featureFlags.rules.customRules
                      </span>
                    </p>
                    <div
                      class="Layout--vertical StyledProps--main"
                    >
                      <div
                        class="Layout--horizontal Layout--layout-spacing-small StyledProps--main"
                      >
                        <div
                          class="StyledProps--font StyledProps--main rulesContainer custom"
                          style="width: 100%;"
                        >
                          <div
                            class="Layout--vertical Layout--layout-spacing-small StyledProps--main"
                          >
                            <p
                              class="StyledProps--font StyledProps--main serveLabel"
                            >
                              cf.featureFlags.serveVariationToTargetLabel
                            </p>
                            <div
                              style="display: flex; align-items: center; justify-content: flex-start;"
                            >
                              <p
                                class="StyledProps--font StyledProps--main"
                              >
                                cf.featureFlags.serve
                              </p>
                              <div
                                style="max-width: 210px;"
                              >
                                <span
                                  style="border-radius: 50%; width: 12px; height: 12px; background-color: rgb(64, 101, 160); display: inline-block; transform: translateY(1px); margin-left: 1px;"
                                />
                                <span
                                  class="StyledProps--font StyledProps--main StyledProps--inline StyledProps--margin-bottom-xsmall"
                                  style="margin-bottom: 0px; font-weight: 600;"
                                >
                                  True
                                </span>
                              </div>
                              <p
                                class="StyledProps--font StyledProps--main"
                              >
                                cf.featureFlags.toTarget
                              </p>
                              <div
                                class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-alignItems-center StyledProps--flex-justifyContent-flex-start"
                              >
                                <div
                                  class="AvatarGroup--stack"
                                  style="--avatar-over-lap: -13px;"
                                >
                                  <div
                                    class="AvatarGroup--stackAvatar Avatar--Avatar Avatar--contentStyle"
                                  >
                                    <span
                                      class="bp3-popover-wrapper Avatar--avatarPopOver"
                                    >
                                      <span
                                        class="bp3-popover-target"
                                      >
                                        <div
                                          class="Avatar--AvatarInner"
                                          style="border-radius: 100%; line-height: 26px; width: 32px; height: 32px; font-size: 10px;"
                                          tabindex="0"
                                        >
                                          A
                                        </div>
                                      </span>
                                    </span>
                                  </div>
                                </div>
                              </div>
                              <p
                                class="StyledProps--font StyledProps--main"
                              >
                                (
                                1
                                )
                              </p>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <button
              class="bp3-button bp3-disabled bp3-minimal Button--button StyledProps--font StyledProps--main btnCode StyledProps--intent-primary Button--with-current-color Button--without-shadow Button--withLeftIcon"
              disabled=""
              tabindex="-1"
              title="cf.featureNotReady"
              type="button"
            >
              <span
                class="bp3-icon bp3-icon-code StyledProps--main"
                icon="code"
              >
                <svg
                  data-icon="code"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <desc>
                    code
                  </desc>
                  <path
                    d="M15.71 7.29l-3-3a1.003 1.003 0 00-1.42 1.42L13.59 8l-2.29 2.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71l3-3c.18-.18.29-.43.29-.71 0-.28-.11-.53-.29-.71zM5 5a1.003 1.003 0 00-1.71-.71l-3 3C.11 7.47 0 7.72 0 8c0 .28.11.53.29.71l3 3a1.003 1.003 0 001.42-1.42L2.41 8 4.7 5.71c.19-.18.3-.43.3-.71zm4-3c-.48 0-.87.35-.96.81l-2 10c-.01.06-.04.12-.04.19 0 .55.45 1 1 1 .48 0 .87-.35.96-.81l2-10c.01-.06.04-.12.04-.19 0-.55-.45-1-1-1z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`FlagActivation it should render form correctly 1`] = `
<div
  class="StyledProps--font StyledProps--main StyledProps--padding-bottom-xlarge StyledProps--padding-left-xlarge StyledProps--padding-right-xlarge"
  id="save-flag-to-git-modal-body"
>
  <h3
    class="StyledProps--font StyledProps--font-h3 StyledProps--main StyledProps--font-variation-h3 StyledProps--margin-bottom-xlarge"
    level="3"
  >
    cf.gitSync.saveFlagToGit
  </h3>
  <div
    class="Layout--vertical Layout--layout-spacing-small StyledProps--main"
  >
    <div
      class="StyledProps--font StyledProps--main"
      style="width: 50%;"
    >
      <div
        class="InputWithIdentifier--txtNameContainer TextInput--main"
      >
        <div
          class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main InputWithIdentifier--txtIdContainer"
        >
          <div
            class="Text--withDocsTooltip"
            data-tooltip-id="flagActivation_flagIdentifier"
          >
            <p
              class="StyledProps--font StyledProps--main"
            >
              ID
            </p>
          </div>
          <p
            class="StyledProps--font StyledProps--main"
          >
            :
          </p>
          <span
            class="bp3-popover-wrapper"
          >
            <span
              class="bp3-popover-target"
            >
              <div
                class="bp3-editable-text bp3-disabled InputWithIdentifier--idValue"
              >
                <span
                  class="bp3-editable-text-content"
                  style="height: 1px; line-height: 1px; min-width: 0;"
                >
                  new_flag
                </span>
              </div>
            </span>
          </span>
        </div>
        <div
          class="bp3-form-group bp3-disabled"
        >
          <label
            class="bp3-label"
            for="flagName"
          >
            <span
              class="Tooltip--acenter"
              data-tooltip-id="flagActivation_flagName"
            >
              name
            </span>
             
            <span
              class="bp3-text-muted"
            />
          </label>
          <div
            class="bp3-form-content"
          >
            <div
              class="bp3-input-group bp3-disabled"
            >
              <input
                autocomplete="off"
                class="bp3-input"
                disabled=""
                name="flagName"
                type="text"
                value="new flag "
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="StyledProps--font StyledProps--main"
    >
      <h4
        class="StyledProps--font StyledProps--font-h4 StyledProps--main StyledProps--font-variation-h5 StyledProps--margin-bottom-small StyledProps--color StyledProps--color-grey600"
        level="4"
      >
        common.gitSync.harnessFolderLabel
      </h4>
      <div
        class="Layout--horizontal Layout--layout-spacing-small StyledProps--main formRow"
      >
        <div
          class="bp3-form-group bp3-disabled"
        >
          <label
            class="bp3-label"
            for="gitDetails.repoIdentifier"
          >
            <span
              class="Tooltip--acenter"
              data-tooltip-id="flagActivation_gitDetails.repoIdentifier"
            >
              common.git.selectRepoLabel
            </span>
             
            <span
              class="bp3-text-muted"
            />
          </label>
          <div
            class="bp3-form-content"
          >
            <div
              class="bp3-input-group bp3-disabled"
            >
              <input
                autocomplete="off"
                class="bp3-input"
                disabled=""
                name="gitDetails.repoIdentifier"
                type="text"
                value="harnesstest"
              />
            </div>
          </div>
        </div>
        <div
          class="bp3-form-group bp3-disabled"
        >
          <label
            class="bp3-label"
            for="gitDetails.rootFolder"
          >
            <span
              class="Tooltip--acenter"
              data-tooltip-id="flagActivation_gitDetails.rootFolder"
            >
              common.gitSync.harnessFolderLabel
            </span>
             
            <span
              class="bp3-text-muted"
            />
          </label>
          <div
            class="bp3-form-content"
          >
            <div
              class="bp3-input-group bp3-disabled"
            >
              <input
                autocomplete="off"
                class="bp3-input"
                disabled=""
                name="gitDetails.rootFolder"
                type="text"
                value="/.harness/"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="bp3-form-group bp3-disabled"
      >
        <label
          class="bp3-label"
          for="gitDetails.filePath"
        >
          <span
            class="Tooltip--acenter"
            data-tooltip-id="flagActivation_gitDetails.filePath"
          >
            common.git.filePath
          </span>
           
          <span
            class="bp3-text-muted"
          />
        </label>
        <div
          class="bp3-form-content"
        >
          <div
            class="bp3-input-group bp3-disabled"
          >
            <input
              autocomplete="off"
              class="bp3-input"
              disabled=""
              name="gitDetails.filePath"
              type="text"
              value="/flags.yaml"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="StyledProps--font StyledProps--main"
    >
      <h4
        class="StyledProps--font StyledProps--font-h4 StyledProps--main StyledProps--font-variation-h5 StyledProps--margin-bottom-small StyledProps--color StyledProps--color-grey600"
        level="4"
      >
        common.gitSync.commitDetailsLabel
      </h4>
      <div
        class="bp3-form-group"
      >
        <label
          class="bp3-label"
          for="gitDetails.commitMsg"
        >
          <span
            class="Tooltip--acenter"
            data-tooltip-id="flagActivation_gitDetails.commitMsg"
          >
            common.git.commitMessage
          </span>
           
          <span
            class="bp3-text-muted"
          />
        </label>
        <div
          class="bp3-form-content"
        >
          <textarea
            autocomplete="off"
            class="bp3-input bp3-fill"
            maxlength="1024"
            name="gitDetails.commitMsg"
            placeholder="common.git.commitMessage"
          />
        </div>
      </div>
      <div
        class="StyledProps--font StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-flex-start StyledProps--flex-alignItems-center"
      >
        <span
          class="bp3-icon StyledProps--main"
          data-icon="git-branch-existing"
        >
          <svg
            fill="none"
            height="16"
            viewBox="0 0 16 9"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M12.403 3.67H16v1.626h-3.597a4.477 4.477 0 01-8.807 0H0V3.67h3.596a4.477 4.477 0 018.807 0zm-7.26.813a2.857 2.857 0 105.714 0 2.857 2.857 0 00-5.714 0z"
              fill="#383946"
              fill-rule="evenodd"
            />
          </svg>
        </span>
        <span
          class="StyledProps--font StyledProps--main StyledProps--margin-left-small StyledProps--margin-right-small StyledProps--inline"
        >
          common.git.existingBranchCommitLabel
          :
        </span>
        <div
          class="bp3-form-group bp3-disabled"
          style="margin-bottom: 0px;"
        >
          <div
            class="bp3-form-content"
          >
            <div
              class="bp3-input-group bp3-disabled"
            >
              <input
                autocomplete="off"
                class="bp3-input"
                disabled=""
                name="gitDetails.branch"
                type="text"
                value="main"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="StyledProps--font StyledProps--main StyledProps--padding-left-xlarge StyledProps--padding-top-small"
        data-testid="commit-details-section"
      >
        <div
          class="bp3-form-group"
        >
          <div
            class="bp3-form-content"
          >
            <label
              class="bp3-control bp3-checkbox bp3-large StyledProps--font Checkbox--checkbox StyledProps--main StyledProps--inline"
            >
              <input
                name="autoCommit"
                type="checkbox"
              />
              <span
                class="bp3-control-indicator"
              />
              <span
                class="Tooltip--acenter FormikForm--checkBoxDocTooltipLabel"
                data-tooltip-id="flagActivation_autoCommit"
              >
                cf.gitSync.autoCommitLabel
              </span>
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="Layout--horizontal Layout--layout-spacing-small StyledProps--main StyledProps--padding-top-xxlarge"
    style="align-items: center;"
  >
    <button
      class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color Button--variation Button--variation-primary"
      data-testid="save-flag-to-git-modal-save-button"
      type="button"
    >
      <span
        class="bp3-button-text"
      >
        save
      </span>
    </button>
    <button
      class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color Button--variation Button--variation-tertiary"
      data-testid="save-flag-to-git-modal-cancel-button"
      type="button"
    >
      <span
        class="bp3-button-text"
      >
        cancel
      </span>
    </button>
  </div>
</div>
`;
